### PR TITLE
fix: add missing step functions permissions

### PR DESCRIPTION
**Pull request by carvantes**
Description of changes:
Fix env variable and add permissions. 
Also created a separate `UpdateStatusLambdaRole` to break a circular dependency( and because it was the right thing to do to begin with)

btw, the `iamRoleStatements` in the `provider` section are added to ALL lambdas unless they provide their own role. That's probably not what we want since those permissions are specific to the `fhirServer` lambda. Still, not changing it right now since that refactor would cause even more conflicts with mainline.

Checklist:

* [x] Have you successfully deployed to an AWS account with your changes?
* [x] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cloudformation/bulkExport.yaml [hunk 1]
```diff
@@ -138,3 +138,32 @@ Resources:
...
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - dynamodb:UpdateItem
+                Resource:
+                  - !GetAtt ExportRequestDynamoDBTable.Arn
+              - Effect: Allow
+                Action:
+                  - kms:Decrypt
```
Comment (rsmayda) [inline cloudformation/bulkExport.yaml:167]: When do we need KMS decryption for a DDB update? I would think encryption is needed?

**Review by rsmayda — COMMENTED** [review]
Good callout:

> btw, the iamRoleStatements in the provider section are added to ALL lambdas unless they provide their own role. That's probably not what we want since those permissions are specific to the fhirServer lambda.

Can you add this as tech debt to our internal tracking system?

File: cloudformation/bulkExport.yaml (hunk 1 above)
Comment (carvantes) [reply cloudformation/bulkExport.yaml:167]: kms:Decrypt is correct. 

You can find the details here: https://docs.aws.amazon.com/kms/latest/developerguide/services-dynamodb.html

tl;dr This allow Is not about decrypting or encrypting DDB table contents. This permissions is needed to decrypt the "table key". DynamoDB internally manages the encryption/decryption of your data.

**Comment by carvantes** [comment]
> Can you add this as tech debt to our internal tracking system?

done

File: cloudformation/bulkExport.yaml (hunk 1 above)
Comment (nguyen102) [reply cloudformation/bulkExport.yaml:167]: @carvantes  thanks for the clarification.

**Review by nguyen102 — APPROVED** [review]
